File: seat-reservation-manager/seat-reservation-manager.java
class SeatManager {

    PriorityQueue<Integer> pq= new PriorityQueue();
    public SeatManager(int n) {
        for(int i=1; i<=n; i++) pq.offer(i);
    }
    
    public int reserve() {
        int seatNumber = pq.poll();
        return seatNumber;
    }
    
    public void unreserve(int seatNumber) {
        pq.offer(seatNumber);
    }
}

// Seat Reservation manager

File: seat-reservation-manager/seat-reservation-manager.py
class SeatManager:

    def __init__(self, n: int):
        self.pq = [i for i in range(1, n+1)]

    def reserve(self) -> int:
        seatNumber = heapq.heappop(self.pq)
        return seatNumber

    def unreserve(self, seatNumber: int) -> None:
        heapq.heappush(self.pq, seatNumber)


# Seat Reservation manager